src/nnbench/runner.py

```diff
@@ -274,6 +282,14 @@ def _maybe_dememo(v, expected_type):
             return v
 
         for benchmark in self.benchmarks:
+            bm_family = benchmark.fn.__name__
+            bm_state = State(
+                name=benchmark.name or bm_family,
```

Comment:
`benchmark.name` should never be None (or ""), so you can leave out the or branch.

Reply:
I had mypy complaining here. But I could get rid of that warning by changeing `if not self.name ` to `if self.name is None` in the benchmark `__post_init__` where we create the name if not set.